packages/block-library/src/block/edit.native.js

```diff
@@ -0,0 +1,262 @@
+/**
+ * External dependencies
+ */
+import { Platform, Text, TouchableWithoutFeedback, View } from 'react-native';
+import { partial } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { Component } from '@wordpress/element';
+import { BottomSheet, Icon, Spinner, Disabled } from '@wordpress/components';
+import { withSelect, withDispatch } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { BlockEditorProvider, BlockList } from '@wordpress/block-editor';
+import { compose, withPreferredColorScheme } from '@wordpress/compose';
+import { parse } from '@wordpress/blocks';
+import { help } from '@wordpress/icons';
+import { requestUnsupportedBlockFallback } from '@wordpress/react-native-bridge';
+
+/**
+ * Internal dependencies
+ */
+import styles from './editor.scss';
+import EditTitle from './edit-title';
+
+class ReusableBlockEdit extends Component {
+	constructor( { reusableBlock } ) {
+		super( ...arguments );
+
+		this.setBlocks = this.setBlocks.bind( this );
+		this.toggleSheet = this.toggleSheet.bind( this );
+		this.requestFallback = this.requestFallback.bind( this );
+
+		if ( reusableBlock ) {
+			// Start in edit mode when we're working with a newly created reusable block
+			this.state = {
+				// Since edition is not supported yet isEditing is always false
+				isEditing: false,
+				blocks: parse( reusableBlock.content ),
+				showHelp: false,
+				sendFallbackMessage: false,
+			};
+		} else {
+			// Start in preview mode when we're working with an existing reusable block
+			this.state = {
+				isEditing: false,
+				blocks: [],
+				showHelp: false,
+				sendFallbackMessage: false,
+			};
+		}
+	}
+
+	componentDidMount() {
+		this.props.fetchReusableBlock();
+	}
+
+	componentDidUpdate( prevProps ) {
+		const hasBlockFetchCompleted =
+			! prevProps.reusableBlock && this.props.reusableBlock;
+		const hasBlockContentChanged =
+			prevProps.reusableBlock?.content !==
+			this.props.reusableBlock?.content;
+
+		if ( hasBlockFetchCompleted || hasBlockContentChanged ) {
+			this.setState( {
+				blocks: parse( this.props.reusableBlock.content ),
+			} );
+		}
```

Comment:
The blocks list is updated when:
1. The reusable block was empty and the fetch completes.
2. The reusable block content has changed (for now this only can happen when is modified through UBE).